File: examples/python/getting-started/custom-boto3-client.py
import boto3
from botocore.config import Config
from aws_durable_execution_sdk_python import durable_execution, DurableContext

# Create a custom boto3 Lambda client with specific configuration
custom_lambda_client = boto3.client(
    'lambda',
    config=Config(
        retries={'max_attempts': 5, 'mode': 'adaptive'},
        connect_timeout=10,
        read_timeout=60,
    )
)

@durable_execution(boto3_client=custom_lambda_client)
def handler(event: dict, context: DurableContext) -> dict:
    # Your durable function logic
    return {"status": "success"}

File: examples/java/getting-started/custom-boto3-client.java
public class MyHandler extends DurableHandler<Map<String, Object>, Map<String, String>> {

    @Override
    protected DurableConfig createConfiguration() {
        // Create a custom Lambda client with specific configuration
        var lambdaClientBuilder = LambdaClient.builder()
            .httpClient(ApacheHttpClient.builder()
                .maxConnections(50)
                .connectionTimeout(Duration.ofSeconds(10))
                .socketTimeout(Duration.ofSeconds(60))
                .build());

        return DurableConfig.builder()
            .withLambdaClientBuilder(lambdaClientBuilder)
            .build();
    }

    @Override
    public Map<String, String> handleRequest(Map<String, Object> event, DurableContext ctx) {
        // Your durable function logic
        return Map.of("status", "success");
    }
}
